frontend/packages/console-dynamic-plugin-sdk/src/extensions/user-preferences.ts

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import { Extension, ExtensionDeclaration, CodeRef } from '../types';
+
+export enum UserPreferencesInputType {
+  dropdown = 'dropdown',
+  checkbox = 'checkbox',
+}
+
+export type DropdownOption = {
+  type: UserPreferencesInputType.dropdown;
+  options: {
+    value: string;
+    label: string;
+  }[];
+};
+export type CheckboxOption = {
+  type: UserPreferencesInputType.checkbox;
+  trueValue: string;
+  falseValue: string;
+};
+export type UserSettingInputOptions = DropdownOption | CheckboxOption;
+
+export type UserSettings = ExtensionDeclaration<
+  'console.user-settings',
```

Comment:
I would suggest to don't call this extension just "UserSettings" as this is a really generic term. Based on the decision if we want call all these types UserPreferences... or UserSettings... I would suggest something `UserPreferenceOption` or `UserPreferenceField`. Wdyt?

```suggestion
export type UserPreferenceField = ExtensionDeclaration<
  'console.user-settings/field',
```

Reply:
@christianvogt Any preference (hihi) here as well? :)

Reply:
It is *item now. Item or field is both for for me.

Reply:
In our call today we decided this extensions should be an item: `UserPreferenceItem`.

The field is the part of an item which describes just the input field (just the checkbox or the dropdown).